frontend/src/components/Header/Header.tsx

```
@@ -1,16 +1,56 @@
+        {/* Right Side: Login (md+) & Theme Toggle */}
+        <div className="flex items-center space-x-4">
+          {/* Login Button (Visible on md+) */}
+          <Button className="hidden md:block rounded-full md:px-6 lg:px-8 xl:px-10">
+            <Link to="/signin">Login</Link>
+          </Button>
+
+          {/* Mode Toggle */}
+          <ModeToggle />
+
+          {/* Mobile Menu Button */}
+          <button className="md:hidden" onClick={() => setIsOpen(!isOpen)}>
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)